vllm/multimodal/base.py

```diff
@@ -55,13 +66,23 @@ def __init__(self) -> None:
                                   MultiModalInputMapper[D]] = {}
 
     @abstractmethod
-    def get_data_type(self) -> Type[D]:
+    def get_internal_data_type(self) -> Type[D]:
```

Comment:
I think this is a bit too convoluted. Let's assume that each key can only support one plugin. That way, we can skip the intermediate step of converting external data type into internal data type, instead converting the external data type into tensor directly via `_default_input_mapper`.

Reply:
yeah the logic is a bit convoluted. I like what you said about "enforce that each key can only correspond to one plugin". But I have some questions when thinking through the details. 

So right now, Registry has some default plugins. Say for "image" type, we put "the_plugin_for_image.image" there. Say there is someone who wants to use the embedding plugin for "image" type, do they have to remove the default plugin and register with the other one?

Reply:
`registry_input_mapper` accepts a custom `mapper` so users can define their own processing on a per-model basis if need be.

Reply:
hmmm, then if someone wants to add a new input type for an existing data key ("image") they need to modify the logic under `ImagePlugin` by adding an elif?

Reply:
They can register their custom input mapper via the optional argument to the decorator